ctpolicy/applepolicy.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ctpolicy
+
+import (
+	"github.com/google/certificate-transparency-go/loglist"
+	"github.com/google/certificate-transparency-go/x509"
+)
+
+// AppleCTPolicy manages policy logic in appliance to the data provided.
+type AppleCTPolicy struct {
+	baseBuilder baseGroupBuilder
+}
+
+// LogsByGroup describes submission requirements according to https://support.apple.com/en-us/HT205280
+func (appleP *AppleCTPolicy) LogsByGroup(cert *x509.Certificate, approved *loglist.LogList) (map[string]*LogGroupInfo, error) {
+	var outerror error
+	groups := make(map[string]*LogGroupInfo)
```

Comment:
Could move this down to the penultimate line so it's only invoked if error checks pass.
```golang
  ...
  groups := map[string]*LogGroupInfo{baseGroup.Name: &baseGroup}
  return groups, nil 
```
or even:
```golang
  ...
  return map[string]*LogGroupInfo{baseGroup.Name: &baseGroup}, nil
```

Reply:
Moved down a bit. Yet, we always return data. Even when have errors.